addons/sale/models/product_product.py

```diff
@@ -18,6 +18,21 @@ class ProductProduct(models.Model):
         search='_search_product_is_in_sale_order',
     )
 
+    @api.model
+    def search_fetch(self, domain, field_names=None, offset=0, limit=None, order=None):
+        partner_id = self.env.context.get('partner_id')
```

Comment:
I would be better to give specific name to this context since to avoid passing this condition where not needed, Line in some product view where partner is needed or somewhere in product M2x field which is getting context from some action or view